cvat/apps/engine/annotation.py

```diff
@@ -82,14 +82,18 @@ def save_job(jid, data):
         .select_for_update().get(id=jid)
 
     annotation = _AnnotationForJob(db_job)
-    annotation.validate_data_from_client(data)
+    client_ids = annotation.validate_data_from_client(data)
 
     annotation.delete_from_db(data['delete'])
     annotation.save_to_db(data['create'])
     annotation.update_in_db(data['update'])
 
     db_job.segment.task.updated_date = timezone.now()
     db_job.segment.task.save()
+
+    db_job.max_shape_id = max(db_job.max_shape_id, max(client_ids['create']) if client_ids['create'] else -1)
```

Comment:
What will has happened if we update or delete any shape and save?
I think db_job.max_shape_id will reset to -1, but it wrong.